enhancements/baremetal/additional-ntp-servers.md

```diff
@@ -0,0 +1,104 @@
+---
+title: additional-ntp-servers
+authors:
+  - "@tdomnesc"
+reviewers:
+  - "@dtantsur"
+approvers:
+  - "@zaneb"
+api-approvers:
+  - None
+creation-date: 2024-11-08
+last-updated: 2024-11-08
+tracking-link:
+  - "https://issues.redhat.com/browse/METAL-1022"
+see-also:
+  - "https://issues.redhat.com/browse/OCPBUGS-22957"
+---
+
+# Support additional NTP servers in install-config that would be passed to IPA
+
+## Summary
+
+When the Metal3 Ironic (https://github.com/openshift/ironic-image) pre-provisioning image containing the Ironic Python Agent (https://github.com/openshift/ironic-agent-image) is running, there is no way to sync the clocks to a custom NTP server. This causes issues with certificates - IPA generates a certificate for itself to be valid starting 1 hour in the past, so if the hardware clock is more than 1 hour ahead of the real time then the certificate will be rejected by Ironic.
```

Comment:
This introduction features a lot of concepts that are specific to Metal platform (Metal3, Ironic, pre-provisioning image, IPA). Ideally, we should approach OCP enhancements from a more generic perspective. So, something like,

"When provisioning the initial cluster using bare-metal IPI installer, or provisioning additional workers using the Metal platform components (Metal3 and Ironic), there is no way to synchronize the hardware clock when the provisioning ramdisk is running. If the clock skew with the bootstrap VM or the control plane is large enough, the temporary TLS certificate generated by the agent inside the ramdisks will be considered either not valid yet or already invalid by Ironic."